spacelift/resource_environment_variable.go

```diff
@@ -82,17 +82,24 @@ func resourceEnvironmentVariable() *schema.Resource {
 				Default:     true,
 				ForceNew:    true,
 			},
+			"description": {
+				Type:        schema.TypeString,
+				Description: "Description of the environment variable",
+				Optional:    true,
+				ForceNew:    true,
```

Comment:
Recreating an environment variable will clear out the value that might have been set through the UI. It may not be evident to users why that's required. It may also require the user to update a secret value, which can be a headache.

Reply:
Should I implement the **UpdateContext** for this field? It seems all other fields in this resource have this property true as well.

Reply:
> Recreating an environment variable will clear out the value that might have been set through the UI.

If an environment variable was created with UI, Terraform won't track it. Users could only import it as data, which is read-only unless I am missing something.

Reply:
> > Recreating an environment variable will clear out the value that might have been set through the UI.
> 
> If an environment variable was created with UI, Terraform won't track it. Users could only import it as data, which is read-only unless I am missing something.

Users sometimes create the env variable through IaC but use the UI to set the value of that variable.

Reply:
> Should I implement the **UpdateContext** for this field?

I believe so. WDYT @tomasmik @Yantrio @silviutanasa?

> It seems all other fields in this resource have this property true as well.

It makes sense to recreate the variable when moving a variable between entities, setting the value, or changing a plain value into a secret. I believe it would be better if we didn't require a recreate when working with a description.

Reply:
In the UI (modification of env var for contexts), the only way to save changes is to change value (neither making as secret or description change can be saved on their own).
<img width="466" alt="Screenshot 2024-08-13 at 10 16 16" src="https://github.com/user-attachments/assets/04ba70c7-042a-439d-a11a-aaa12e1c46ad">

If the value is modified, the resource will get [recreated](https://github.com/spacelift-io/terraform-provider-spacelift/blob/6ef7b8791f3ad38628a0a6209324fdb78c4fa66b/spacelift/resource_environment_variable.go#L76).

Reply:
Recreating the variable makes it consistent with the UI.